File: src/org/irislang/jiris/compiler/IrisRunnable.java
package org.irislang.jiris.compiler;

import org.irislang.jiris.core.IrisContextEnvironment;
import org.irislang.jiris.core.IrisThreadInfo;
import org.irislang.jiris.core.IrisValue;

public interface IrisRunnable {
	public abstract IrisValue run(IrisContextEnvironment context, IrisThreadInfo threadInfo);
}

File: src/main/java/org/irislang/jiris/IrisRunnable.kt
package org.irislang.jiris

import org.irislang.jiris.core.IrisContextEnvironment
import org.irislang.jiris.core.IrisThreadInfo
import org.irislang.jiris.core.IrisValue

interface IrisRunnable {
    fun run(context: IrisContextEnvironment, threadInfo: IrisThreadInfo): IrisValue
}
